### PR TITLE
Lose rep on eval prop fail, new onperiod

### PR DESCRIPTION
**Pull request by IanMendozaJaimes**
## Issues
#265 
#232 

## Contracts
Onboarding
Proposals
Settings

updateSettings

## Migration

- This assumes the pull request #262 has been merged

This is the feature branch, the migration branch is: **migration/lose-rep-on-fail-eval**. In the migration branch there are two methods:
- `migcycstats`: passes the data from (old) 'cyclestats' to migration table 'mcyclestats', this method does not erase any entry of 'cyclestats'
- `delcycstats`: this method erases the table `cyclestats`

The next step is this branch, there is a method called `initcycstats`, this method passes the information from `mcyclestats` to (new) `cyclestats`

**This is a new way to run onperiod, it uses a new index in the proposal table, so this shouldn't be merged before #262**

It also includes the issue #232

### CODE REVIEW

File: src/seeds.accounts.cpp [hunk 1]
```diff
@@ -484,6 +484,8 @@ void accounts::addrep(name user, uint64_t amount)
     });
   }
 
+  print("ADDDING REP: ", user, ", amount=", amount, "\n");
```
Comment (n13) [inline src/seeds.accounts.cpp:487]: Not needed, can go away

File: src/seeds.proposals.cpp [hunk 2]
```diff
@@ -261,237 +261,369 @@ bool proposals::is_active(name account, uint64_t cutoff_date) {
...
+    bool is_alliance_type = prop_type == alliance_type;
+    bool is_campaign_type = prop_type == campaign_type;
+
+    bool valid_quorum = false;
+
+    if (pitr->status == status_evaluate) { // in evaluate status, we only check unity. 
+      valid_quorum = true;
+    } else { // in open status, quorum is calculated
+      valid_quorum = utils::is_valid_quorum(voters_number, quorum, total_eligible_voters);
+    }
```
Comment (n13) [inline src/seeds.proposals.cpp:360]: Need to merge new quorum calculation into this

File: src/seeds.proposals.cpp [hunk 3]
```diff
@@ -261,237 +261,369 @@ bool proposals::is_active(name account, uint64_t cutoff_date) {
...
-    // }
+  auto props_by_stage = props.get_index<"bystage"_n>();
 
-    // print("smallest id: "+std::to_string(smallest_prop_id) );
-      
-    // auto pitr = props.find(smallest_prop_id);
-    auto pitr = props.find(50);
+  auto spitr = props_by_stage.find(stage_staged.value);
+  while (spitr != props_by_stage.end() && spitr->stage == stage_staged) {
+    send_eval_prop(spitr->id, c.propcycle);
```
Comment (n13) [inline src/seeds.proposals.cpp:500]: I would create a separate method for eval

Comment (n13) [reply src/seeds.proposals.cpp:500]: In any case we can put that in another PR - maybe leave this as is, one change at a time!

**Review by n13 — APPROVED** [review]
Approved with comment

See comments ;)

File: src/seeds.proposals.cpp (hunk 3 above)
Comment (IanMendozaJaimes) [reply src/seeds.proposals.cpp:503]: I agree with you, another function for just staged proposals is a good idea, but I also think that could go into another pull request

**Comment by IanMendozaJaimes** [comment]
- Merged with master, removed the comment in accounts

**Comment by n13** [comment]
Merged master into this, unit tests working

**Comment by n13** [comment]
Migration on mainnet complete

**Comment by n13** [comment]
- [ ] Onboarding
- [ ] Proposals
- [ ] Settings
- [ ] updateSettings
- [ ]

**Comment by n13** [comment]
Deployed

- [x] Onboarding
- [x] Proposals
- [x] Settings
- [x] updateSettings

**Pull request merged**